mpp/include/mpp_update_domains2D.h

```diff
@@ -227,6 +227,7 @@
       pointer( ptr_in,  field3D_in  )
       pointer( ptr_out, field3D_out )
 
+      field_out = 0
```

Comment:
I don't know about this one...
1.  Are you sure this is the compiler behavior for all compilers?  Will this change answers?
2. This code is very confusing with the pointers.  Are you sure this isn't somehow getting set in mpp_redistribute?

Reply:
It shouldn't change answers unless the answers are dependent on uninitialized values. It most likely is getting set in mpp_redistribute, it's just good practice to set to some initial value and gets rid of the warnings.

Reply:
You're right, this code *should have* initialized `field_out` because it is an `intent(out)` variable.  I completely agree with everything you said, but this is going to need extensive testing to make sure it's not changing answers.  I wouldn't be surprised if there was some code somewhere using the uninitialized values, and initializing with 0 changes the answers.  It's done in multiple places, which gives multiple chances for answer changes.